### DISCUSSION TITLE
Faders value and banks

### DISCUSSION THREAD

**Discussion by pmjcontroller1** [shanteacontrols/OpenDeck#288] [General]
Hi paradajz,
We have detected an issue with the faders values and banks.
Until now, when you move the fader to a value in bank 1, for example to 25%, and click to another bank 2, and move the fader to 50%, when comeback to the bank 1, the midi output didn't send value until the fader engage to the 25%. But now, when you come to bank 1, the fader sends value immediately. 
That's a problem because the midi apps become a bit crazy.
Thanks!!!

**Comment by paradajz** [comment]
What is bank?

**Reply by pmjcontroller1** [reply]
Sorry I would mean Presets

**Comment by paradajz** [comment]
And what do you mean 'but now'? After some firmware version?

**Comment by pmjcontroller1** [comment]
Good morning Paradajz

We have just tested from version 7.5.4, which is the one we started working with.

We were wrong, it doesn't work the way we thought it should.

The logical operation would be to store the current values ​​in each preset and change the preset, load the memorized values ​​and not send data until the faders, buttons, etc., reach the memorized value.

Example.
We configure the pcb with 2 presets 2 faders (
We start the controller with all the faders at 0 in all preset.

We move fader 1 preset 1 to 50%
We change to preset 2,
The moment we move fader 1 it starts sending data from the 50% position

The logical behavior we believe should be, it does not send anything until it reaches 0%, which is its stored position in this case.

Thank you

**Reply by paradajz** [reply]
I will check but please stop writing "Paradax" when the name is clearly "paradajz"

**Reply by pmjcontroller1** [reply]
Sorry Paradajz

**Comment by paradajz** [comment]
> We move fader 1 preset 1 to 50%
We change to preset 2,
The moment we move fader 1 it starts sending data from the 50% position

This is a very logical and intended behaviour.

> The logical behavior we believe should be, it does not send anything until it reaches 0%

Well I do not agree at all, the code works exactly as it should. I don't understand why would you want what you've written.

**Comment by basst22778** [comment]
I understand the problem:

If you control a DAW, the DAW saves the last value and only changes it when the incoming value is greater than the saved value (pickup mode). The controller doesn't need any logic for that.

However, if you want to control a hardware synth where the same four potentiometers (potis) are used to control two different envelopes (e.g., preset 1 - EG 1 and preset 2 - EG 2), then the pickup of the values would need to be implemented in the controller.

Alternatively, you could buy an Opendeck L and use discrete potentiometers for each value.

**Reply by paradajz** [reply]
For encoders there is an option "remote sync" where the CC value of encoder is synced with the one DAW sends. So the same software pickup mechanism could be implemented for analog potentiometers, I guess.

**Reply by pmjcontroller1** [reply]
in the lighting software we use, like Grandma.  The software itself stores its values.  What we need is for opendeck to also store them and respect them when changing presets.  so that they only vary once they reach the stored memory.  If not, there is a problem when returning from preset and moving a potentiometer that is not at its identical value.  Each knob has a different MIDI to control different destinations.  I don't know if I can explain myself.


Tomorrow I will make a video that is surely more explanatory and you will be able to see what I am trying to explain.

**Comment by pmjcontroller1** [comment]
Hi paradajz
in the below link you can find a explanaition of the concept that we discuss. The example shows the behavior of faders but we think that it must be in all controls. The idea is maintain the current state/value in the preset that it is working. When the preset changes, the values must be indepently from the other preset. Then the sent values for this current preset, only must be sent when the physical position match with the logical value (engage it). 

https://drive.google.com/drive/folders/1cx5g2UEth9GyL1dvsSuufv1R6A4qTauX?usp=sharing

**Reply by paradajz** [reply]
When you switch to preset 2 in your software, which messages are sent to the board?

**Reply by pmjcontroller1** [reply]
none, no feedback

**Reply by paradajz** [reply]
But how do you switch the preset on OpenDeck then?

**Reply by pmjcontroller1** [reply]
Since there is no communication between the two, as is the case with many software, we need OpenDeck to do its independent work so that they remain synchronized.  There is no need to send updates.  only opendeck stores its preset values and only sends value changes when the fader reaches the previous stored value

**Reply by paradajz** [reply]
You misunderstood me. How are you switching presets in OpenDeck?

**Reply by pmjcontroller1** [reply]
We have 4 buttons for preset selection.  By clicking on them we choose the preset.  In each preset we have configured the fader and the buttons with the same MIDI note but a different channel.
